devito/passes/iet/languages/openacc.py

```diff
@@ -164,8 +164,7 @@ def _make_partree(self, candidates, nthreads=None, index=0):
         if self._is_offloadable(root) and \
            all(i.is_Affine for i in [root] + collapsable) and \
            self.par_tile:
-            idx = min(index, len(self.par_tile) - 1)
-            tile = self.par_tile[idx]
+            tile = self.par_tile.next()
```

Comment:
See your comment below, basically allow to avoid corner case where the user assumes the number of loop nest wrong so it reuses the last one if there is missing inputs.